content/swift/basic-operators-in-swift.md

```diff
@@ -0,0 +1,198 @@
+
+The Closed Range operator(`a...z`) creates a range that begins at `a` and continues to and includes, `z`. The most common use case for Closed Range Operators is in `for-in` loops.
+
+```swift
+let userAge = 32
+for year in 2...userAge {
+  print("You turned \(year) years old!")
+}
+// prints:
+// "You turned 2 years old!"
+// "You turned 3 years old!"
+// ... all the way to and including
+// "You turned 32 years old!"
+```
+
+Half-Open range operators(`a..<z`) create a range beginning at `a` and continue to `z` but not including `z`. They are commonly used when working with Arrays or other zero-indexed lists.
+
+```swift
+let picnic = ["Ceasar Salad", "Turkey Sandwich", "Cookie", "Juice Box"]
+for i in 0..<picnic.count {
+  print("Item #\(i+1) in the picnic basket is a \(picnic[i]).")
+}
+// Item 1 in the picnic basket is a Ceasar Salad.
+// Item 2 in the picnic basket is a Turkey Sandwiches.
+// Item 3 in the picnic basket is a Cookie.
+// Item 4 in the picnic basket is a Juice Box.
+```
+
+The One-Sided range operator defines a range that will continue as far in one direction as possible. We can use this range to iterate from or to specific ranges inside of other fixed length entities.
+
+```swift
+for food in picnic[...2] {
+  print(food)
+}
+// Ceasar Salad
+// Turkey Sandwich
+// Cookie
+
+for food in picnic[2...] {
+  print(food)
+}
+// Cookie
+// Juice Box
+
+for food in picnic[..<3] {
+  priont(food)
+}
+// Ceasar Salad
+// Turkey Sandwich
+// Cookie
+
+let infiniteRange = 0... // 0 to infinity
+for i in infiniteRange {
+  print(i)
+  // creates an infinite loop, always take note of what
+  // "stops" your open-ended ranges
+}
+```
+
+## Nil-Coalescing Operator
+
+The nil-coalescing operator (`a ?? z`) unwraps an optional with the value of `a` unless `a` is nil, then it will return the value of `z`. `a` always has to be an optional type and `z` has to match the type that's stored in `a`.
+
+```swift
+let defaultPrice = 1.99
+var getItemPrice: Double?
+// Since getItemPrice isn't assigned a value it defaults to 'nil'
+// The price to charge the customer is either a price we get from an item
+// or the default price. In this case, priceToChargeCustomer = 1.99
+let priceToChargeCustomer = getItemPrice ?? defaultPrice
+```
+
+## Ternary Conditional Operator
+
+The ternary conditional operator is shorthand for a simple `if-else` statement. It follows the syntax `statementToTest ? doThisAndReturnIfTrue : doThisAndReturnIfFalse` which in terms of the `if-else` statement can be read as: `if` the statement is true, evaluate and return the value of the left side of the ternary, if it's false, evaluate and return the value of the right side.
```

Comment:
`if-else` -> `if`-`else`

and

- `if` the statement is true, evaluate and return the value of the left side of the ternary.
- `if` it's false, evaluate and return the value of the right side.